### ISSUE TITLE
Tab navigation issues: Bad visibility and illogical order

### ISSUE THREAD

**Issue by cgruber-uos** [polimediaupv/paella-ethz#9]
**Bad visibility**

![2022-12-12 14_54_29-Greenshot](https://user-images.githubusercontent.com/101632236/207063026-c9f67685-d6e1-476f-99bb-005d5b069a38.png)

On the sceenshot the second icon is active, you see a slight blue line around the icon and the icon is a little bigger than the left one. The contrast and visibility of an active tabed element is too weak. We should use double colored borders (that work on light and dark background) and / or a color that is clearly distinguishable from the button color! 

**Illogical order** 

The tab order on the player needs a rework: At the moment the upper icons (in den silde and video) are the first one (1) then the player UI icons (2) are navigatable. So you have to tab to all icons above to go to the essentials UI elements like play / pause / forward / backward: 

![2022-12-12 14_43_47-Greenshot](https://user-images.githubusercontent.com/101632236/207060249-121f6df5-24f2-40c4-978d-e4467d50e0b6.png)

We should change this: After a fresh loaded player page the fist tabbing should go to area 2! 

Also in the area tab order is Illogical 

![2022-12-12 14_36_34-Greenshot](https://user-images.githubusercontent.com/101632236/207061633-e05a7864-1b87-46e5-969d-c533ad7b4a42.png)

and should be re-orderd so that the visual order should match the tab order in general. 

Same in the main navigation: Currently you jump from play to settings and then back to backward / forward /

![2022-12-12 14_50_04-Greenshot](https://user-images.githubusercontent.com/101632236/207061995-b783cd04-b71f-4aed-82a4-7c447c115047.png)
![2022-12-12 14_50_18-Greenshot](https://user-images.githubusercontent.com/101632236/207062009-6ed1cd3d-b213-4f6a-85d8-2a0e37af57d8.png)
![2022-12-12 14_50_19-Greenshot](https://user-images.githubusercontent.com/101632236/207062030-88fefe2f-f854-4a2a-b23a-4da95e0f098a.png)

**Comment by cgruber-uos** [comment]
In addition: The volume slider is never accessible by tabbing. You can only mute the video but the volume slider does never show up by using the tab key only:

![2022-12-15 09_53_47-Greenshot](https://user-images.githubusercontent.com/101632236/207816850-096e8426-bd6a-4ffe-91e1-9b642310cf8e.png)

Other players (with fading in / out volume slider) do have this functionality, e.g. YouTube: 

![2022-12-15 09_53_25-Greenshot](https://user-images.githubusercontent.com/101632236/207816892-08bd8ef4-dd6a-4f80-ab27-7c5bb400c33d.png)

**Comment by miesgre** [comment]
@cgruber-uos We updated the player. Can you test it works as expected?
Demo: https://polimediaupv.github.io/paella-ethz/?id=belmar-multiresolution-remote

**Comment by ClemensGruber** [comment]
Thanks, much better order now! Still some issues: 

- the volume slider is still not accessible via keyboard only, by focusing the loudspeaker I can press enter and mute or unmute but I have no opportunity to adjust the volume because the slider is not apperearing. 
- the blue border for a focused element is good visible on the grey background but nearly not visible on the video / slides "inline" buttons. They are also blue (same color!!) so the icons appears just slight bigger than the other - in case you know it! :-) 

![2023-03-02 23_07_16-Greenshot](https://user-images.githubusercontent.com/36139413/222571024-9a3cf1e6-cc37-4d23-b8c6-b4cae35bb20a.png)

vs. 

![2023-03-02 23_07_34-Greenshot](https://user-images.githubusercontent.com/36139413/222571037-4a464d26-8512-403c-b522-a3ce12fa000a.png)

Are the colors set or can we change the border color or - perhaps a better solution - can we add an additional slight light border (on the video / slide inline buttons only) inside the focus border to make focused more prominent

**Comment by miesgre** [comment]
I changed de border color of element selected.
![Captura de pantalla 2023-03-03 a las 9 34 17](https://user-images.githubusercontent.com/2735202/222675125-dcb8158f-c255-4b1a-92e5-005768dc0593.png)

I changed for all buttons (slide inline and playbar).
@oas777 What do you think? maybe lisa has a better option

Demo: https://polimediaupv.github.io/paella-ethz/?id=belmar-multiresolution-remote

**Comment by oas777** [comment]
Well, before we ask Lisa, we probably should ask Clemens whether this is the best option for accessibility. Because if it is, the design is less important...

**Comment by miesgre** [comment]
Yes. Clemens suggested to change the border of the buttons of the slides because accessibility.
Clemens say the blue border button in the play bar is correct.

But If we change the border in slide buttons, should we change also in playbar buttons? or we are going to have two color border (one for playbar buttons and another for slide buttons)?

Also we need to choose the border color. I selected a "pink" border to test. But I'm not sure if its the best color. In this Clemens can help us.

**Comment by ClemensGruber** [comment]
I think we should find a good solution for a11y and desing. So the pink color would not be my favourite (with the desing hat :-) And I would stay with the former color in the navigation area and an other in the video, in case different imlementaion it is not too costly. I would try a dark blue for the inline buttons. 

Miguel / @miesgre can you set - for testing only - a dark blue for tab focus, just to see how it woks. In case it is ok we can implement different colors for navigation and video / slides. 

This would be not totally consistant, because we would use two different colors for the same effect. But I think it works because in the navigation are we have a mor "dark theme" (because of the background color) and in the video a "light theme", but let's try!

**Comment by miesgre** [comment]
I changed the border to a dark blue color (slides and playbar)

**Comment by ClemensGruber** [comment]
No, bad idea from my side, too less contrast! Pink?! Olaf / Lisa any idea for a color that fits both?

**Comment by miesgre** [comment]
@oas777 Seems tab navigation issues are resolved. What about the focus border color (with keyboard)? Pink is good? Lisa prefers another color?
Demo: https://polimediaupv.github.io/paella-ethz/?id=belmar_16_9_hls

Playbar buttons:
![Captura de pantalla 2023-04-19 a las 10 20 41](https://user-images.githubusercontent.com/2735202/233014091-c6d91988-e9d8-4795-a93c-8e3544ca8536.png)

Video buttons:
![Captura de pantalla 2023-04-19 a las 10 21 01](https://user-images.githubusercontent.com/2735202/233014156-2430a2c3-ac65-471d-8b8a-24f41a115df8.png)

**Comment by cgruber-uos** [comment]
A bit ot: 

> Demo: https://polimediaupv.github.io/paella-ethz/?id=belmar_16_9_hls

Is the demo URL working on your side? I got only a never ending loading icon after clicking on the preview image: 

![2023-04-19 11_01_35-Greenshot](https://user-images.githubusercontent.com/101632236/233025190-35e4ffa3-455a-4724-84bf-d5fa44b99699.png)

**Comment by miesgre** [comment]
There is a problem with Safari and CORS. I will try to fix it. Meanwhile you can use chrome (maybe edge too)


> A bit ot:
> 
> > Demo: https://polimediaupv.github.io/paella-ethz/?id=belmar_16_9_hls
> 
> Is the demo URL working on your side? I got only a never ending loading icon after clicking on the preview image:
> 
> ![2023-04-19 11_01_35-Greenshot](https://user-images.githubusercontent.com/101632236/233025190-35e4ffa3-455a-4724-84bf-d5fa44b99699.png)

**Comment by oas777** [comment]
Clemens, good to see you are still here...

As for the color of the border: The designer would ask for blue, the accessibility expert calls for pink: Can we meet in purple? `#800080`

**Comment by miesgre** [comment]
> 

Changed to purple, but maybe it does not have enough contrast?
![Captura de pantalla 2023-04-19 a las 13 33 29](https://user-images.githubusercontent.com/2735202/233062968-e05af359-0671-485a-a4e9-ab5e2147faa3.png)

**Comment by oas777** [comment]
Clemens?

**Comment by oas777** [comment]
PS: https://polimediaupv.github.io/paella-ethz/?id=belmar_16_9_hls works on Edge and Chrome, but not on Firefox (all Windows). Will test on OS X later.

**Comment by cgruber-uos** [comment]
> PS: https://polimediaupv.github.io/paella-ethz/?id=belmar_16_9_hls works on Edge and Chrome, but not on Firefox (all Windows). Will test on OS X later.

Earlier version of the test page did work on FF, I had no issue with all test via https://polimediaupv.github.io/paella-ethz/... till now. so it must be a newly introduced but or a change in FF? 

> the accessibility expert calls for pink:

I fear pink was introduced by chance. :-) No reason beside contrast. 

> Changed to purple, but maybe it does not have enough contrast?
> ![Captura de pantalla 2023-04-19 a las 13 33 29](https://user-images.githubusercontent.com/2735202/233062968-e05af359-0671-485a-a4e9-ab5e2147faa3.png)

And this is the problem with purple: Less contrast. Not only with the dark blue in the navigation bar, also with the light blue buttons in the video / slide area. Miguel / @miesgre can you play a bit with the color, perpaps a trial could be a light purple to get a higher contrast on both elements inline buttons above and navigation buttons below.

**Comment by oas777** [comment]
Seems correct to point the finger: OS X works with Safari (!) and Chrome - but not on Firefox.

**Comment by miesgre** [comment]
> > PS: https://polimediaupv.github.io/paella-ethz/?id=belmar_16_9_hls works on Edge and Chrome, but not on Firefox (all Windows). Will test on OS X later.
> 
> Earlier version of the test page did work on FF, I had no issue with all tests via https://polimediaupv.github.io/paella-ethz/... till now. so it must be a newly introduced bug or a change in FF.
> 

It should now work on Firefox. It was a server misconfiguration.

**Comment by oas777** [comment]
Yes (for Windows), thanks.

**Comment by miesgre** [comment]
> Miguel / @miesgre can you play a bit with the color, perpaps a trial could be a light purple to get a higher contrast on both elements inline buttons above and navigation buttons below.

I updated to use 20% lighter purple.

Other colors we can use:
`#800080` purple
`#cc00cc` 20% lighten
`#f200f2` 30% lighten
`#ff19ff` 40% lighten
`#ff40ff` 50% lighten

@oas777, @cgruber-uos Opinions?

**Comment by oas777** [comment]
I let Clemens decide.

**Comment by cgruber-uos** [comment]
In the video the background color of the buttons is `#3a8bdd`
In the navigation bar the background color is `#1f2937`

#### Results WCAG Contrast checker

`#cc00cc` (20% lighten) 
- background `#3a8bdd` vs. `#cc00cc` (20% lighten)   
-> contrast ratio: 1.33, did **neither** fit AA **nor** AAA level requirements 
- background `#1f2937` vs. `#cc00cc` (20% lighten)   
-> contrast ratio: 3.1, fits **AA for big elements** but nut AAA 

`#f200f2` (30% lighten) 
- background `#3a8bdd` vs. `#f200f2` (30% lighten)   
-> contrast ratio: 1.02, did **neither** fit AA **nor** AAA level requirements 
- background `#1f2937` vs. `#f200f2` (30% lighten)   
-> contrast ratio: 4.23, fits **AA for big elements** but nut AAA 

And so on. That means we will never get a matching color for both, the contrast with the dark background and the contrast with the light backgrund. 

I wold try `#f200f2` (30% lighten) so the contrast in the main navigation is high and would fit good with a darker video background but not so good with a light video.

**Comment by miesgre** [comment]
Changed to `#f200f2`

**Comment by miesgre** [comment]
@oas777 @cgruber-uos ?
I think the only thing left to do is to choose the color of the border.
I changed the border color to `#f200f2`. 
<img width="206" alt="Captura de pantalla 2023-05-04 a las 19 14 10" src="https://user-images.githubusercontent.com/2735202/236277633-ee129eae-2cf2-4034-b7bd-fc79d8d0df71.png">
<img width="222" alt="Captura de pantalla 2023-05-04 a las 19 14 26" src="https://user-images.githubusercontent.com/2735202/236277657-98694421-555f-4701-8965-565b87939c5c.png">


Is it ok? Can we close this issue?

**Comment by oas777** [comment]
Let's wait a week for Clemens to answer (he's not working for the Opencast project anymore), otherwise we'll stick with the color and can close this.

**Comment by cgruber-uos** [comment]
Not the optimum but ok I would say!

**Comment by oas777** [comment]
That's all we wanted to hear...
